core/state-machine/src/overlayed_changes.rs

```diff
@@ -48,22 +68,307 @@ pub struct OverlayedValue {
 	pub extrinsics: Option<BTreeSet<u32>>,
 }
 
-/// Prospective or committed overlayed change set.
-#[derive(Debug, Default, Clone)]
+/// History of value that are related to a state history (eg field `history` of
+/// an `OverlayedChangeSet`).
+///
+/// First field is the current size of the history.
+/// Second field are statically allocated values for performance purpose.
+/// Third field are dynamically allocated values.
```

Comment:
I am not sure what these comments refer to?

Can we introduce a nominal type with explicit fields?

Reply:
My mistake, I miss the documentation update when switching to smallvec.